### DISCUSSION TITLE
Interacting with Server

### DISCUSSION THREAD

**Discussion by athyfr** [Infinidoge/nix-minecraft#100] [Q&A] [answered]
I am sorry if this is supposed to be common knowledge with NixOS or systemd, but I couldn't find any answers anywhere else.

How can I access the server log / run commands as the server? I couldn't find anything on how to do this.

When you normally run a Minecraft server (without services), it outputs all sorts of logs as the plugins load and as players join etc. and also allows running Minecraft commands from the terminal. I can't find this interface.

Help, please!

**Comment by Infinidoge** [comment, marked as answer]
Heya! To interact with the server,
- If using tmux: `tmux attach -S /run/minecraft/server-name.sock`
-If using systemd sockets: `echo some command > /run/minecraft/servername.stdin`

Make sure you are in the `minecraft` group (or use `sudo`), otherwise you won't have the permissions to access the socket!

**Reply by no-mood** [reply]
> Make sure you are in the `minecraft` group (or use `sudo`), otherwise you won't have the permissions to access the socket!

I think this could be referenced in the readme/module, since new (linux) users may not know :)

**Comment by athyfr** [comment]
Alright, thanks!
I'll be sure to try that out, and then I can get the error messages to report another issue.
